src/product-catalog/main.go

```diff
@@ -178,6 +177,43 @@ type productCatalog struct {
 	pb.UnimplementedProductCatalogServiceServer
 }
 
+func loadProductCatalog() {
+	log.Info("Loading Product Catalog...")
+	var err error
+	catalog, err = readProductFiles()
+	if err != nil {
+		log.Fatalf("Error reading product files: %v\n", err)
+		os.Exit(1)
+	}
+
+	// Default reload interval is 10 seconds
+	interval := DEFAULT_RELOAD_INTERVAL
+	si := os.Getenv("PRODUCT_CATALOG_RELOAD_INTERVAL")
+	if si != "" {
+		interval, _ = strconv.Atoi(si)
+		if interval <= 0 {
+			interval = DEFAULT_RELOAD_INTERVAL
+		}
+	}
+	log.Infof("Product Catalog reload interval: %d", interval)
+
+	ticker := time.NewTicker(time.Duration(interval) * time.Second)
+
+	go func() {
+		for {
```

Comment:
It would be best to have the programs' context here so it can be properly shutdown, but it seems it would require a major refactor to the codebase (all other init functions as well). We can keep it for a future improvement.

Reply:
This is something I think we can do in a future improvemenbt